conversion/conversion.go

```diff
@@ -387,11 +409,20 @@ func dataFromDatabaseForDataflowMigration(targetProfile profiles.TargetProfile,
 		if err != nil {
 			return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
 		}
+
+		// Fetch and store the GCS bucket associated with the datastream
+		dsClient := getDatastreamClient(ctx)
+		gcsBucket, _, fetchGcsErr := streaming.FetchTargetBucketAndPath(ctx, dsClient, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg.DestinationConnectionConfig)
+		if fetchGcsErr != nil {
+			logger.Log.Error(fmt.Sprintf("Could not fetch GCS Bucket for Shard %s hence Monitoring Dashboard will not contain Metrics for the gcs bucket\n error=%v\n", p.DataShardId, fetchGcsErr))
+		}
+
+		// create monitoring dashboard for a single shard
 		monitoringResources := metrics.MonitoringMetricsResources{
 			ProjectId:            targetProfile.Conn.Sp.Project,
 			DataflowJobId:        dfOutput.JobID,
 			DatastreamId:         streamingCfg.DatastreamCfg.StreamId,
-			GcsBucketId:          streamingCfg.TmpDir,
+			GcsBucketId:          gcsBucket,
 			PubsubSubscriptionId: streamingCfg.PubsubCfg.SubscriptionId,
 			SpannerInstanceId:    targetProfile.Conn.Sp.Instance,
 			SpannerDatabaseId:    targetProfile.Conn.Sp.Dbname,
```

Comment:
Curious - we don't need `region` to chart some of these resources?

Reply:
https://cloud.google.com/storage/docs/buckets
It mentions in this documentation that the name needs to be "globally unique"